### PR TITLE
MLB-1740: Fix Circle CI Pipeline to work with sdk 35 (Android 15)

### PR DESCRIPTION
**Pull request by jlplks**
# 📲 What

Fix CircleCI Pipeline to work with SDK 35(ANDROID 15

# 🤔 Why

CircleCI pipeline broke after using an Android Image that has SDK 35

# 🛠 How

- Make some changes in CircleCI configuration to make it use JVM 17 on the build
- updated some dependencies
- fix tests

### CODE REVIEW

**Review by Arkariang — APPROVED** [review]
💖  nicely done!

**Comment by jlplks** [comment]
> 💖 nicely done!

Thanks for the help!

File: app/src/main/java/com/kickstarter/viewmodels/ThreadViewModel.kt [hunk 1]
```diff
@@ -407,7 +406,10 @@ interface ThreadViewModel {
             /** reversed replies **/
             apolloPaginate
                 .paginatedData()
-                ?.map { it.reversed() }
+                ?.map {
+                    it.toMutableList().reverse()
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/ThreadViewModel.kt:410]: fyi @mtgriego could be related with the issues you found when migrating Discovery to RXJava2

Comment (mtgriego) [reply app/src/main/java/com/kickstarter/viewmodels/ThreadViewModel.kt:410]: ill look into it, thanks!

**Pull request merged**